core/src/test/resources/ExplicitImplicit/basic.source

```diff
@@ -44,14 +71,124 @@ class A {
 class A {
   class B { class C }
   implicit val x = new B
-  implicit val y = new x.C
+  implicit val y: x.C = new x.C
 }
 >>>
 class A {
   class B { class C }
   implicit val x: B = new B
   implicit val y: x.C = new x.C
 }
+<<< NOWRAP higher kinded cats
+package cats {
+  package laws {
+    package discipline {
+      trait CartesianTests[F[_]]
+      object CartesianTests {
+        trait Isomorphisms[F[_]]
+        object Isomorphisms {
+          def id[T[_]]: Isomorphisms[T] = ???
+        }
+      }
+    }
+  }
+}
+package cats {
+  trait Id[T]
+}
+package cats {
+  package tests {
+    import cats.laws.discipline._
+    class IdTests {
+      implicit val iso = CartesianTests.Isomorphisms.id[Id]
+    }
+  }
+}
+>>>
+package cats {
+  package laws {
+    package discipline {
+      trait CartesianTests[F[_]]
+      object CartesianTests {
+        trait Isomorphisms[F[_]]
+        object Isomorphisms {
+          def id[T[_]]: Isomorphisms[T] = ???
+        }
+      }
+    }
+  }
+}
+package cats {
+  trait Id[T]
+}
+package cats {
+  package tests {
+    import cats.laws.discipline._
+    class IdTests {
+      implicit val iso: CartesianTests.Isomorphisms[Id] = CartesianTests.Isomorphisms.id[Id]
+    }
+  }
+}
+<<< higher kinded
+import scala.concurrent.Future
```

Comment:
No, this line is needed. Have a look at the shortened `Future` in the implicit val type result.